tests/WE2E/run_WE2E_tests.py

```diff
@@ -0,0 +1,447 @@
+                       "has not been specified in the machine file for this platform")
+    if not os.path.isdir(mach['platform']['TEST_EXTRN_MDL_SOURCE_BASEDIR']):
+        raise FileNotFoundError(dedent(f"""The directory for staged test data specified in this platform's machine file
+                                TEST_EXTRN_MDL_SOURCE_BASEDIR = {mach['platform']['TEST_EXTRN_MDL_SOURCE_BASEDIR']}
+                                does not exist."""))
+
+    # Different input data types have different directory structures, so set the data directory accordingly
+    if cfg_lbcs['EXTRN_MDL_NAME_LBCS'] == 'FV3GFS':
+        if 'FV3GFS_FILE_FMT_LBCS' not in cfg_lbcs:
+            cfg_lbcs['FV3GFS_FILE_FMT_LBCS'] = dflt['task_get_extrn_lbcs']['FV3GFS_FILE_FMT_LBCS']
+        cfg_lbcs['EXTRN_MDL_SOURCE_BASEDIR_LBCS'] = f"{mach['platform']['TEST_EXTRN_MDL_SOURCE_BASEDIR']}/"\
+                                                    f"{cfg_lbcs['EXTRN_MDL_NAME_LBCS']}/{cfg_lbcs['FV3GFS_FILE_FMT_LBCS']}/${{yyyymmddhh}}"
+    else:
+        cfg_lbcs['EXTRN_MDL_SOURCE_BASEDIR_LBCS'] = f"{mach['platform']['TEST_EXTRN_MDL_SOURCE_BASEDIR']}/"\
+                                                    f"{cfg_lbcs['EXTRN_MDL_NAME_LBCS']}/${{yyyymmddhh}}"
+
+    return cfg_lbcs
+
+def setup_logging(logfile: str = "log.run_WE2E_tests", debug: bool = False) -> None:
+    """
+    Sets up logging, printing high-priority (INFO and higher) messages to screen, and printing all
+    messages with detailed timing and routine info in the specified text file.
+    """
+    logging.getLogger().setLevel(logging.DEBUG)
+
+    formatter = logging.Formatter("%(name)-16s %(levelname)-8s %(message)s")
+
+    fh = logging.FileHandler(logfile, mode='w')
+    fh.setLevel(logging.DEBUG)
+    fh.setFormatter(formatter)
+    logging.getLogger().addHandler(fh)
+
+    logging.debug(f"Finished setting up debug file logging in {logfile}")
+    console = logging.StreamHandler()
+    if debug:
+        console.setLevel(logging.DEBUG)
+    else:
+        console.setLevel(logging.INFO)
+    logging.getLogger().addHandler(console)
+    logging.debug("Logging set up successfully")
+
+
+
+if __name__ == "__main__":
+
+    # Check python version and presence of some non-standard packages
+    check_python_version()
+
+    #Get the "Home" directory, two levels above this one
+    homedir=os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+    logfile='log.run_WE2E_tests'
+
+    #Parse arguments
+    parser = argparse.ArgumentParser(epilog="For more information about config arguments (denoted in CAPS), see ush/config_defaults.yaml\n")
+    optional = parser._action_groups.pop() # Create a group for optional arguments so they can be listed after required args
+    required = parser.add_argument_group('required arguments')
+
+    required.add_argument('-m', '--machine', type=str, help='Machine name; see ush/machine/ for valid values', required=True)
```

Comment:
@mkavulich The "machine" argument needs to be made lower-case before being used since this failed:
```
./run_WE2E_tests.py --machine Hera --account rtrr --tests MET_verification --expt_basedir set01
```
with
```
FileNotFoundError: [Errno 2] No such file or directory: '/scratch2/BMC/det/Gerard.Ketefian/UFS_CAM/TEST_mkavulich_new_run_WE2E_tests/ufs-srweather-app/ush/machine/Hera.yaml'
```
but changing "Hera" to "hera" worked.

Reply:
Fixed, this was needed a few other places too.